src/index.tsx

```diff
@@ -20,122 +25,136 @@ export default class SignatureCanvas extends Component {
     clearOnResize: PropTypes.bool
   }
 
-  static defaultProps = {
+  static defaultProps: Pick<ReactSignatureCanvasProps, 'clearOnResize'> = {
     clearOnResize: true
   }
 
-  _sigPad = null
+  // this is some hack-ish init and casting to avoid `| null` everywhere :/
+  /* eslint-disable @typescript-eslint/consistent-type-assertions */
+  _sigPad: SignaturePad = {} as SignaturePad
+  _canvas: HTMLCanvasElement = {} as HTMLCanvasElement
+  /* eslint-enable @typescript-eslint/consistent-type-assertions */
+
+  _setRef = (ref: HTMLCanvasElement | null): void => {
+    if (ref !== null) {
+      this._canvas = ref
+    }
+  }
 
-  _excludeOurProps = () => {
+  _excludeOurProps = (): SignaturePad.SignaturePadOptions => {
```

Comment:
Instead of prefixing private properties or methods, `public` and `private` TS modifiers can be used.

Reply:
Yea I was thinking of doing that now that it's in TS, but I don't yet totally know all the ramifications or transpilation differences of `private` or the new `#` modifier

Reply:
Will be leaving this as is for now so that's it's not breaking any use of these that may have occurred. In particular, I've [used and mentioned using `_resizeCanvas`](https://github.com/agilgur5/react-signature-canvas/issues/46#issuecomment-595592861) in the past.

Moving forward, should use JS-style `#` for private fields